tools/image_processing/yolo/yolo_prediction.xml

```diff
@@ -0,0 +1,102 @@
+<tool id="yolo_predict" name="YOLO Prediction" version="@VERSION@+galaxy0" profile="24.2">
+    <description>Prediction with YOLO models</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <requirements>
+	<container type="docker" >quay.io/sunyi000/ultralytics-galaxy:latest</container>
+    </requirements>
+    <command detect_errors="aggressive">
+    <![CDATA[
+      export YOLO_CONFIG_DIR='\$HOME/.config/ultralytics' &&
+
+      mkdir -p ./input_images ./runs ./models ./results && 
+
+      #for $i, $filename in enumerate($input_images):
+          ln -s '$filename' './input_images/${filename.element_identifier}' &&
+      #end for
+
+      cp '$model' './models/model.pt' &&
+
+      python '$__tool_directory__/yolov8.py' 
+		    --test_path='./input_images' 
+		    --model_path='./models' 
+		    --model_name='model' 
+		    --run_dir='./runs' 
+		    --save_dir='./results'
+		    --image_size='$image_size'
+		    --mode='$mode'
+		    --foldername='overlaid_images'
+		    --scale='$scale'
+		    --degree='$degree'
+		    --class_names_file='$class_name'
+		    --confidence='$confidence'
+		    --iou='$iou'
+		    --max_det='$max_det'  
+
+    ]]>
+    </command>
+    <inputs>
+        <param name="input_images" type="data" format="jpg,png,tiff" multiple="true"  label="Input images"/>
+        <param name="num_class" type="integer" label="Number of class names for images" />
+	<param name="class_name" type="data" format="txt" label="YOLO class name file" />
+	<param name="model" type="data" format="data" label="Model file" />
+	<param name="mode" type="select" label="Prediction mode">
+              <option value="segment">segment</option>
+              <option value="detect">detect</option>
+	</param>
+	<param name="image_size" type="integer" value="1000" label="Image size" help="Size of input image to be used only as integer of w,h." />
+	<param name="scale" type="float" value="0.5" label="Image scale" help="Image scale (+/- gain)" />
+	<param name="degree" type="float" value="0.0" label="Image rotation" help="Image rotation (+/- deg)" />
+ 	<param name="confidence" type="float" value="0.5" label="Confidence" help="Confidence value (0-1) for each detected bounding box" />
+	<param name="iou" type="float" value="0.7" label="IoU" help="Intersection over Union threshold for NMS" />
+	<param name="max_det" type="integer" value="300" label="Max. of detections" help="Maximum number of detections allowed per image. Limits the total number of objects the model can detect in a single inference, preventing excessive outputs in dense scenes." />
+    </inputs>
+    <outputs>
+	 <collection name="results" type="list" label="YOLO prediction results">
```

Comment:
I think it would be better to have two separate outputs and not one collection with two heterogenous elements.

Reply:
just checked here, the no. of prediction result files depends.
for example, 
if the no. of images to predict is 10, the result no. of output files will be 20 (10masks, 10 txt)
maybe better to have collection here..

for the tests in this yolo_prediction.xml, image to predict is just one, that's why it only has 2 outputs.